### DISCUSSION TITLE
A tree-sitter grammar

### DISCUSSION THREAD

**Discussion by mtoohey31** [koka-lang/koka#370] [General]
Hello!

Just wanted to mention that I've created a tree-sitter grammar for Koka. Here's the repository: https://github.com/mtoohey31/tree-sitter-koka. It parses _almost_ all of the standard library without error.

A handful of editors support tree-sitter, some out of the box (here's a list: https://github.com/tree-sitter/tree-sitter/issues/1678), and others via extensions. I've got a PR open for Helix [here](https://github.com/helix-editor/helix/pull/8727). I might create PRs for other editors at some point, but if anyone else wants to beat me to it, feel free!

Here's a screenshot of the syntax highlighting in Helix with the gruvbox theme:

![koka](https://github.com/koka-lang/koka/assets/36740602/d3651719-2561-4e8f-a937-241666896be6)

**Comment by akirak** [comment]
@mtoohey31 Thank you! I have opened a PR (https://github.com/NixOS/nixpkgs/pull/267000) to add the grammar to NixOS. If it is approved, I will probably work on an Emacs major mode that uses the grammar.
